### ISSUE TITLE
Spot-diagram data

### ISSUE THREAD

**Issue by DrPaprikaa** [mjhoptics/ray-optics#16]
Hi, 

Is there a way to access the spot-diagram data at specified field/wavelength/zoom positions ?
It would be ideal for image quality evaluation purposes : plotting, data analysis...

Thank you,
DrPaprikaa

**Comment by mjhoptics** [comment]
I have added a notebook to the ray-optics-notebooks repo on GitHub. This sets up a tilted parabola and calculates spot diagrams and wavefront errors several different ways. It also shows how to use jupyter widgets to set up a defocus and tilt dashboard with sliders to make changes and update the plots. The second part uses RayList data sources to colelct the ray data for plotting and other statistical calculations.
I should make this into a video, but it will take a while.
Please let me know if you have additional questions.

**Comment by DrPaprikaa** [comment]
Hi,
Thank you, I've been playing with it a bit.
To better illustrate my issue, I am working with a Ritchey-Chretien telescope, which is diffraction limited on the optical axis. Since I cannot add the system manually (`add_surface()` or `add_mirror()` raise errors), I imported it from a personal .seq file :

![Cassegrain](https://user-images.githubusercontent.com/64958936/102516849-9f2d9e00-408f-11eb-8a46-2ea4389c009e.png)

The command `opm.ele_model.list_elements()` gives the following info :
```
0: M1 (Mirror): Mirror: Conic(c=-0.00025, cc=-1.018159509202454), t=20.0131, sd=400.2622
1: AirGap M1-M1 (AirGap): Gap(t=-1630.0, medium=Air())
2: M1 (Mirror): Mirror: Conic(c=-0.0010810810810810809, cc=-2.441717791411043), t=4.7838, sd=95.6770
3: AirGap M1-Image (AirGap): Gap(t=1850.0, medium=Air())
4: Image (DummyInterface): Surface(lbl='Img', profile=Spherical(c=0.0), interact_mode=transmit)
```

I'm interested in the spot-diagram data, in order to compare the results with CodeV. Using :
```
layout_plt = plt.figure(FigureClass=InteractiveLayout, opt_model=opm, do_draw_rays=True,
                        do_paraxial_layout=False,
                        is_dark=False).plot()
```

I get the following plot :

![Spot_diagram](https://user-images.githubusercontent.com/64958936/102517442-53c7bf80-4090-11eb-8955-fb1f14ca3f46.png)

I have 3 questions : 

1. Can the plot be prettier ?
2. How can I plot the non-normalized spot data, instead of the normalized one ?
3. How can I have acces to a list of polychromatic spot-diagram (x, y) coordinates for a field, to compute the polychromatic RMS spot-size ?

Please excuse me if such functionnalities are already implemented !

Kindly,
DrPaprikaa

**Comment by mjhoptics** [comment]
Hello DrPaprikaa,
I'm sorry you're still having issues. And I apologize for not mentioning the name of the notebook: 'Off-axis parabola notebook v3.0.ipynb'
I worked up a script with your example: 
[Ritchey-Chretien_example.py.zip](https://github.com/mjhoptics/ray-optics/files/5717748/Ritchey-Chretien_example.py.zip)

This shows how to trace a ray list at each field, plot a spot diagram and calculate the RMS spot size for each field. I included using the SpotDiagramFigure with a user scale to control the display of the data. The Jupyter notebook environment displays plots in a way (too small) that I haven't been able to figure out how to change. Hopefully the other options give you more control over the display.

**Comment by nheske** [comment]
Hi mjhoptics,

In the example zip and 'Off-axis parabola notebook v3.0.ipynb',  the call to RayGeoPSF: 

RayGeoPSF(spotfig, gs[:,:1], ray_list, user_scale_value=us, scale_type='user', dsp_typ='hist2d',
                 yaxis_ticks_position='left', title='Geometrical PSF', cmap='gray')

the first two don't seem to match the RayGeoPSF signature:
(ray_list, user_scale_value=0.1, scale_type='fit', yaxis_ticks_position='left', dsp_typ='hist2d', **kwargs)

Is there a newer version of RayGeoPSF constructor or am I missing something?

thanks

**Comment by mjhoptics** [comment]
Hello @nheske,
Yes, there has been a change to the API. I updated the 'Off-axis parabola notebook v3.0.ipynb' notebook and several others to use the new protocol. They are in my mjhoptics/ray-optics-notebooks repo.
I wanted to make these capabilities easier to set up for "simple" cases and was not happy with how the individual plots had to know how to place themselves inside the larger figure. The downside is I broke existing scripts and notebooks.
The changes are in rayoptics v0.6.4 and involve the rayoptics.mpl.analysisfigure and rayoptics.raytr.analyses modules.
Sorry for the confusion.
